From 45daa6901ddccce367c5c0d3eb6d796dbb963f29 Mon Sep 17 00:00:00 2001
From: Adam Duskett <adam.duskett@amarulasolutions.com>
Date: Tue, 5 Aug 2025 11:51:20 +0200
Subject: [PATCH] use root instead of uuid in kernel cmdline

This single line change forces android boot to use root= when booting
instead of the part uuid. The reason for this change is mender sets
root=/dev/mmcblk0pX. Without this change, the root= line is ignored,
causing /dev/mmcblk0p3 to always be the root partition when booting.

Signed-off-by: Adam Duskett <adam.duskett@amarulasolutions.com>
---
 arch/arm/mach-rockchip/board.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-rockchip/board.c b/arch/arm/mach-rockchip/board.c
index e5445af739..a562276412 100644
--- a/arch/arm/mach-rockchip/board.c
+++ b/arch/arm/mach-rockchip/board.c
@@ -1377,7 +1377,7 @@ static void bootargs_add_dtb_dtbo(void *fdt, bool verbose)
 		 * high priority system to boot and add its UUID
 		 * to cmdline. The format is "roo=PARTUUID=xxxx...".
 		 */
-#ifdef CONFIG_ANDROID_AB
+#ifndef CONFIG_ANDROID_AB
 		env_update_filter("bootargs", bootargs, "root=");
 #else
 		env_update("bootargs", bootargs);
-- 
2.50.1

